test/Microsoft.Azure.WebJobs.Host.UnitTests/AttributeClonerTests.cs

```diff
@@ -140,6 +141,35 @@ public string TemplateBind(PropertyInfo propInfo, Attribute attribute, BindingTe
             }
         }
 
+        // Validation with AutoResolve
+        [Binding]
+        public class ValidationWithAutoResolveAttribute : Attribute
+        {
+            [RegularExpression("a+")]
+            [AutoResolve]
+            public string Value { get; set; }
+
+        }
+
+        // Validation with AppSetting
+        [Binding]
+        public class ValidationWithAppSettingAttribute : Attribute
+        {
+            [RegularExpression("a+")]
+            [AppSetting]
+            public string Value { get; set; }
+        }
+
+        // Validation with neither AppSetting nor AutoResolve
+        [Binding]
+        public class ValidationOnlyAttribute : Attribute
+        {
+            // Allow  { } that look like token substitution, but it's not since this isn't AutoResolve
+            [RegularExpression("^.{1,3}$")] 
```

Comment:
I think we should also consider honoring these validation attributes when applied to poco bindings. Allowing declarative validation to be applied to function inputs.